src/helpers/git.js

```diff
@@ -24,9 +24,21 @@ function parseSlug(slug) {
 function parseSlugFromRemoteAddr(remoteAddr) {
   let slug = ''
   if (!remoteAddr) {
-    remoteAddr = childProcess.execSync(
-      "git config --get remote.origin.url || hg paths default || echo ''",
-    )
+    remoteAddr = childProcess
+      .spawnSync('git', [
+        'config',
+        '--get',
+        'remote.origin.url',
+        '||',
+        'hg',
+        'paths',
+        'default',
+        '||',
+        'echo',
+        "''",
+      ])
```

Comment:
This will have to be split up into two separate calls to spawnSync:
- Try to call `git`
- Fallback to `hg`

Lines 28 to 38 are arguments passed to Git. If spawnSync was given `'bash'` instead of `'git'` and `'git'` was placed at the start of the array, this kind of pattern would work but it would be the equivalent of calling execSync.

Reply:
@thomasrockhu do we have any metrics on what percentage of our customers use mercurial? I'm very tempted to drop support for it here.

Reply:
> EDIT: I hope that paints a better picture of how spawnSync and execFileSync differ from execSync.

It does, thanks. I was suspicious that would fail, I've never encountered mercurial so was not sure how to test.

Reply:
unfortunately we don't @drazisil, it would be worth adding that metadata

Reply:
Created https://github.com/codecov/uploader/issues/183